filecoin-proofs/src/api/mod.rs

```diff
@@ -535,106 +524,92 @@ pub unsafe extern "C" fn get_staged_sectors(
     raw_ptr(response)
 }
 
-#[repr(C)]
-pub struct FFISectorClass {
-    sector_size: FFISectorSize,
-    seal_proof_partitions: FFISealProofPartitions,
-    post_proof_partitions: FFIPoStProofPartitions,
-}
-
-#[allow(non_camel_case_types)]
-#[derive(Clone, Debug, Copy)]
-#[repr(C)]
-pub enum FFISectorSize {
-    SSB_OneKiB,
-    SSB_TwoHundredFiftySixMiB,
-}
-
-#[allow(non_camel_case_types)]
-#[repr(C)]
-pub enum FFISealProofPartitions {
-    SPP_Two,
-}
-
-#[allow(non_camel_case_types)]
-#[repr(C)]
-pub enum FFIPoStProofPartitions {
-    PPP_One,
+unsafe fn try_into_post_proofs_bytes(
+    proof_partitions: u8,
+    flattened_proofs_ptr: *const u8,
+    flattened_proofs_len: libc::size_t,
+) -> crate::error::Result<Vec<Vec<u8>>> {
+    post_proof_partitions::try_from_u8(proof_partitions)
+        .map(PoStProofBytesAmount::from)
+        .map(usize::from)
+        .map(|chunk| into_proof_vecs(chunk, flattened_proofs_ptr, flattened_proofs_len))
 }
 
-impl From<FFISectorSize> for SectorSize {
-    fn from(x: FFISectorSize) -> Self {
-        match x {
-            FFISectorSize::SSB_OneKiB => SectorSize::OneKiB,
-            FFISectorSize::SSB_TwoHundredFiftySixMiB => SectorSize::TwoHundredFiftySixMiB,
-        }
-    }
+unsafe fn try_into_porep_proof_bytes(
+    proof_partitions: u8,
+    flattened_proofs_ptr: *const u8,
+    flattened_proofs_len: libc::size_t,
+) -> crate::error::Result<Vec<u8>> {
+    porep_proof_partitions::try_from_u8(proof_partitions)
+        .map(PoRepProofBytesAmount::from)
+        .map(usize::from)
+        .map(|chunk| into_proof_vecs(chunk, flattened_proofs_ptr, flattened_proofs_len))
+        .and_then(|vs| {
+            vs.first()
```

Comment:
In fact, now that I reach this point, it looks as though we are discarding all but the first of the proof vectors returned by `into_proofs_vec`. *If* `into_proofs_vec` had indeed been called with the correct number of partitions (which I now hypothesize it was *not*) — then we would be discarding partitions. Instead, we're simply discarding non-existent extra partitions. This is because we've pretended `proof_chunks` was the size of all the chunks combined, when it should have just been the (constant) size of one proof partition.

'Fortunately', this all works out 'fine' because `verify_seal` does all the work of splitting a larger proof vector into correct-sized chunks anyway. So none of these machinations bought us anything. In fact, we introduced and canceled out errors — perhaps somewhat an over-reliance on the assumption that at the end of the day what type checks must be correct.